### PR TITLE
refactor(ValueMap)

### PR DESCRIPTION
**Pull request by asuper0**
ValueMap::index use &, avoid clone for FieldNames

### CODE REVIEW

**Comment by asuper0** [comment]
The change makes custom method like
```rust
    fn hander(_val: &mut ValueMap) -> Result<(), String> {}
```
unable to implement for the lifetime problem. I don't know how to implement FromValue for ValueMap.

**Comment by tu6ge** [comment]
used to some fields value into handler, e.g.: password and password confirm is same or not

File: src/value/mod.rs [hunk 1]
```diff
@@ -177,22 +180,24 @@ impl Value {
...
-        Some(value)
-    }
-}
+// impl<'b> FromValue for ValueMap<'b> {
+//     fn from_value<'a>(value: &'a mut ValueMap) -> Option<&'a mut Self>
+//     where 'b:'a
+//     {
+//         Some(value)
+//     }
+// }
```
Comment (tu6ge) [inline src/value/mod.rs:189]: Important steps should not lose

**Review by tu6ge — CHANGES_REQUESTED** [review]
lose necessary implementation

File: src/value/mod.rs (hunk 1 above)
Comment (asuper0) [reply src/value/mod.rs:189]: impl `FromValue` for `ValueMap` can't be done due to trait `Rult`. Close this PR.

**Pull request closed without merging**